### PR TITLE
Update NuGet packages to fill out all metadata.

### LINKED ISSUE #43: Fill out nupkg metadata completely
### System information

- **OS version/distro**:  All
- **.NET Version (eg., dotnet --info)**:  All

### Issue

- **What did you do?**

Inspect the NuGet metadata for Microsoft.ML

- **What happened?**

Only some of the info is filled out.  For example, for "description" it says "Package Description".  It also doesn't have a link to release notes or license/project URL.

- **What did you expect?**

I expected all the info to be filled out.

### LINKED ISSUE #103: SourceLink doesn't work with the Microsoft.ML NuGet package
### System information

- **OS version/distro**: Windows
- **.NET Version (eg., dotnet --info)**: net471

### Issue

- **What did you do?**
Debug an application that uses Microsoft.ML.
Load the symbols from the public Microsoft symbols server.
Try to F11 into `new LearningPipeline();`.

- **What happened?**
Visual Studio prompted me to find the LearningPipeline.cs file myself.

- **What did you expect?**
I expected Visual Studio to automatically download the file from GitHub for me and step into it.

### PR DESCRIPTION
**Pull request by eerhardt**
Also, a minor build change (move property ordering) to fix SourceLink with our packages.

Fix #43
Fix #103

### CODE REVIEW

File: pkg/Directory.Build.props [hunk 1]
```diff
@@ -7,6 +7,15 @@
     <IncludeSymbols>false</IncludeSymbols>
   </PropertyGroup>
 
+  <!-- nuspec properties -->
+  <PropertyGroup>
+    <Authors>Microsoft</Authors>
+    <PackageLicenseUrl>https://github.com/dotnet/machinelearning/blob/master/LICENSE</PackageLicenseUrl>
+    <PackageProjectUrl>http://dot.net/ml</PackageProjectUrl>
+    <PackageIconUrl></PackageIconUrl>
```
Comment (eerhardt) [inline pkg/Directory.Build.props]: I still need to get this URL from @shauheen.  My plan is to put a fwlink or aka.ms link here that points to the icon somewhere on the web.  For the .NET Core packages, we point our fwlink to Gravatar.

File: pkg/Microsoft.ML.Parquet/Microsoft.ML.Parquet.nupkgproj [hunk 2]
```diff
@@ -2,8 +2,7 @@
 
   <PropertyGroup>
     <TargetFramework>netstandard2.0</TargetFramework>
-    <NoBuild>true</NoBuild>
-    <IncludeBuildOutput>false</IncludeBuildOutput>
+    <PackageDescription>A set of Apache Parquet components for ML.NET.</PackageDescription>
```
Comment (eerhardt) [inline pkg/Microsoft.ML.Parquet/Microsoft.ML.Parquet.nupkgproj]: Looking for better wording, if anyone has some feedback.

File: pkg/Directory.Build.props [hunk 3]
```diff
@@ -7,6 +7,15 @@
...
+    <PackageReleaseNotes>https://github.com/dotnet/machinelearning/tree/master/Documentation/release-notes</PackageReleaseNotes>
```
Comment (eerhardt) [inline pkg/Directory.Build.props]: I have 2 suggested solutions to the package release notes problem listed here:

https://github.com/dotnet/corefx/issues/29314#issuecomment-387898104

I am taking the first approach for now (pointing to the root of the release-notes folder).  If we agree on the templating approach, we can change it.

File: pkg/Microsoft.ML.Parquet/Microsoft.ML.Parquet.nupkgproj (hunk 2 above)
Comment (TomFinley) [reply pkg/Microsoft.ML.Parquet/Microsoft.ML.Parquet.nupkgproj]: I suppose if I had to try to improve it, I'd point out that these aren't "Apache Parquet components" (I don't know that there's any such animal as that) but rather "ML.NET components". So maybe: "ML.NET components for Apache Parquet support." It's a bit shorter in addition to being, I suspect, slightly more precise.

---
In reply to: [187427255](https://github.com/dotnet/machinelearning/pull/115#discussion_r187427255) [](ancestors = 187427255)

Comment (eerhardt) [reply pkg/Microsoft.ML.Parquet/Microsoft.ML.Parquet.nupkgproj]: Updated.

File: pkg/Directory.Build.props [hunk 4]
```diff
@@ -7,6 +7,15 @@
     <IncludeSymbols>false</IncludeSymbols>
   </PropertyGroup>
 
+  <!-- nuspec properties -->
+  <PropertyGroup>
```
Comment (TomFinley) [inline pkg/Directory.Build.props]: >PropertyGroup [](start = 3, length = 13)

Just curious, does anything here in the nuspec make it clear to consumers of the nuget not only what repository it comes from, but what the commithash was on build? Not sure if that's buried somewhere in here, if it was it wasn't obvious.

File: pkg/Microsoft.ML.Parquet/Microsoft.ML.Parquet.nupkgproj (hunk 2 above)
Comment (TomFinley) [reply pkg/Microsoft.ML.Parquet/Microsoft.ML.Parquet.nupkgproj]: It's not big deal though IMHO

---
In reply to: [187460198](https://github.com/dotnet/machinelearning/pull/115#discussion_r187460198) [](ancestors = 187460198,187427255)

**Comment by glebuk** [comment]
LGTM

File: pkg/Directory.Build.props (hunk 4 above)
Comment (eerhardt) [reply pkg/Directory.Build.props]: There is a new NuGet feature with the `2.1.200` SDK that just got [merged](https://github.com/dotnet/machinelearning/pull/105) that will put the commit hash and repo in the nuspec file.

I'll update this PR to set these properties as well.

See these random collection of links about this:
* https://github.com/NuGet/Home/issues/6547
* https://github.com/NuGet/NuGet.Client/pull/2081
* https://github.com/dotnet/designs/blob/master/accepted/diagnostics/source-link.md
* https://github.com/dotnet/designs/blob/master/accepted/build/standard-ci-env-variables.md

Comment (TomFinley) [reply pkg/Directory.Build.props]: Very nice, thanks @eerhardt .

**Pull request merged**